server/app/get_data.py

```diff
@@ -14,30 +17,118 @@ def get_executions(mlmdfilepath, pipeline_name):
 
 
 # This function fetches all the artifacts available in given mlmd
-def get_artifacts(mlmdfilepath, pipeline_name,data):
+def get_artifacts(mlmdfilepath, pipeline_name, data):
     query = cmfquery.CmfQuery(mlmdfilepath)
     names = query.get_pipeline_names()  # getting all pipeline names in mlmd
     identifiers = []
     for name in names:
-        if name==pipeline_name:
-            stages = query.get_pipeline_stages(name)
-            for stage in stages:
-                executions = query.get_all_executions_in_stage(stage)
-                dict_executions = executions.to_dict("dict")  # converting it to dictionary
-                identifiers.append(dict_executions['id'][0])
-
+        stages = query.get_pipeline_stages(name)
+        for stage in stages:
+            executions = query.get_all_executions_in_stage(stage)
+            dict_executions = executions.to_dict("dict")  # converting it to dictionary
+            identifiers.append(dict_executions["id"][0])
     name = []
     url = []
     df = pd.DataFrame()
     for identifier in identifiers:
-        get_artifacts = query.get_all_artifacts_for_execution(identifier)  # getting all artifacts
-        artifacts_dict = get_artifacts.to_dict('dict')  # converting it to dictionary
+        get_artifacts = query.get_all_artifacts_for_execution(
```

Comment:
we need to run performance measurement to check the most ideal way of querying. Another approach would be to get all artifacts in the store and filter them by pipeline name. we need to measure the performance at scale